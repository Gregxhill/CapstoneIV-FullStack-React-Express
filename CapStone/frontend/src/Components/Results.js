import React from "react";
import { Link } from "react-router-dom";

export default function Results({ isLoading, userInfo, gitBucket }) {
  // This component renders the results of the search

  return (
    // if the loading state is false it indicates that the data has been retrieved and it is then formatted and returned.
    <>
      {!isLoading ? (
        <>
          <div className="github">
            <h1>GitHub:</h1>
            {/* This Link component routes the user to another page which renders the User component with the corresponding user data */}
            <Link to={{ pathname: "/user", state: { userInfo: userInfo } }}>
              <h2>{userInfo[0].login}</h2>{" "}
            </Link>
            <img src={userInfo[0].avatar_url} alt="github avatar" />{" "}
          </div>
          <div className="gitlab">
            <h1>GitLab:</h1>
            {/* If the GitBucket data is empty it indicates that the user doesnt have a GitBucket account. If it isnt empty the corresponding data is rendered. */}
            {gitBucket.length !== 0 ? (
              <h2>{userInfo[1][0].name}</h2>
            ) : (
              <p>User not on GitLab</p>
            )}
            {gitBucket.length !== 0 ? (
              <img src={userInfo[1][0].avatar_url} alt="avatar_gitbucket" />
            ) : null}
          </div>
        </>
      ) : null}
      {/* If the data has been retrieved and it has an error message property, it indicates that the user doesnt exist and the error message is rendered */}
      {userInfo.message && !isLoading ? <h1>{userInfo[0].message}</h1> : null}
    </>
  );
}
