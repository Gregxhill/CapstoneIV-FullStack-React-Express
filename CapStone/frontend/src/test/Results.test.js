import React from "react";
import Results from "../Components/Results";
import renderer from "react-test-renderer";

//this is a snapshot test when the component isn't sent data
test("renders correctly with no data", () => {
  const tree = renderer
    .create(<Results isLoading={true} userInfo={[]} gitBucket={[]} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

//this is a snapshot test when the component is sent the user data
test("renders correctly with data", () => {
  const tree = renderer
    .create(
      <Results
        isLoading={false}
        userInfo={[
          {
            login: "Gregxhill",
            id: 78414802,
            node_id: "MDQ6VXNlcjc4NDE0ODAy",
            avatar_url: "https://avatars.githubusercontent.com/u/78414802?v=4",
            gravatar_id: "",
            url: "https://api.github.com/users/Gregxhill",
            html_url: "https://github.com/Gregxhill",
            followers_url: "https://api.github.com/users/Gregxhill/followers",
            following_url:
              "https://api.github.com/users/Gregxhill/following{/other_user}",
            gists_url: "https://api.github.com/users/Gregxhill/gists{/gist_id}",
            starred_url:
              "https://api.github.com/users/Gregxhill/starred{/owner}{/repo}",
            subscriptions_url:
              "https://api.github.com/users/Gregxhill/subscriptions",
            organizations_url: "https://api.github.com/users/Gregxhill/orgs",
            repos_url: "https://api.github.com/users/Gregxhill/repos",
            events_url:
              "https://api.github.com/users/Gregxhill/events{/privacy}",
            received_events_url:
              "https://api.github.com/users/Gregxhill/received_events",
            type: "User",
            site_admin: false,
            name: "Greg Hill",
            company: null,
            blog: "www.linkedin.com/in/gregory-hill-webdvlpr",
            location: "South Africa",
            email: null,
            hireable: null,
            bio: "Aspiring full-stack web developer and Info Management hons graduate.",
            twitter_username: null,
            public_repos: 6,
            public_gists: 0,
            followers: 0,
            following: 0,
            created_at: "2021-02-02T10:09:47Z",
            updated_at: "2021-08-04T15:47:56Z",
          },
          [],
        ]}
        gitBucket={[]}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
