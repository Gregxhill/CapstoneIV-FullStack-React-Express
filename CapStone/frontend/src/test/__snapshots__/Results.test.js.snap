// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with no data 1`] = `null`;
